test/unittest/component/test_graph.py

```diff
@@ -0,0 +1,123 @@
+from test.db_config import DBConfig
+
+import networkx as nx
+import pytest
+
+from superduperdb.components.graph import Graph
+from superduperdb.components.model import Model
+
+
+@pytest.fixture
+@pytest.mark.parametrize("db", [DBConfig.mongodb_empty], indirect=True)
+def model1(db):
+    def model_object(x):
+        return x + 1
+
+    model = Model(identifier='m1', object=model_object)
+    db.add(model)
+    yield model
+
+
+@pytest.fixture
+@pytest.mark.parametrize("db", [DBConfig.mongodb_empty], indirect=True)
+def model2(db):
+    def model_object(x):
+        return x + 2, x
+
+    model = Model(identifier='m2', object=model_object)
+    db.add(model)
+    yield model
+
+
+@pytest.fixture
+@pytest.mark.parametrize("db", [DBConfig.mongodb_empty], indirect=True)
+def model2_multi(db):
+    def model_object(x, y):
+        return x + y + 2
+
+    model = Model(identifier='m2_multi', object=model_object)
+    db.add(model)
+    yield model
+
+
+@pytest.fixture
+@pytest.mark.parametrize("db", [DBConfig.mongodb_empty], indirect=True)
+def model3(db):
+    def model_object(x, y):
+        return x + y + 3
+
+    model = Model(identifier='m3', object=model_object)
+    db.add(model)
+    yield model
+
+
+def test_simple_graph(model1, model2):
+    g = Graph(identifier='simple-graph', input=model1)
+    g.connect(model1, model2)
+    assert g.predict(1, one=True) == (4, 2)
+
+    g = Graph(identifier='simple-graph', input=model1)
+    g.connect(model1, model2)
+    assert g.predict([1, 2, 3], one=False) == [(4, 2), (5, 3), (6, 4)]
+
+
+def test_complex_graph(model1, model2_multi, model3, model2):
```

Comment:
Is it necessary to support how to get the value from the dictionary output by the upstream node in the future?

For example: 
```
{"confidence": 0.9, "label": 1}
```

I want to get the "label"

Or extract a slice from the result, for example top k items of the result

Reply:
Ah great suggestion, we can add it In next version probably, I had similar idea to let user index outputs 
Lets create an issue for this Thanks